src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_k8s_linux.go

```diff
@@ -72,3 +80,164 @@ func executeK8SBundle(ctx context.Context, bp bundleParams) error {
 	fmt.Printf("Debug bundle saved to %q\n", f.Name())
 	return nil
 }
+
+// adminAddressesFromK8S returns the admin API host:port list by querying the
+// K8S Api.
+func adminAddressesFromK8S(ctx context.Context) ([]string, error) {
+	// This is intended to run only in a k8s cluster:
+	k8sCfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get kubernetes cluster configuration: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(k8sCfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create kubernetes client: %v", err)
+	}
+
+	// Get pods in the 'redpanda' namespace.
+	pods, err := clientset.CoreV1().Pods("redpanda").List(ctx, v1.ListOptions{})
```

Comment:
Can this file be tested? We might need to punt tests to the next release, but I just want to be sure we track it.

Reply:
Sure, If it's fine to tackle this in a separate PR, I'll create an issue to track this. We can use our current k8s test suite.

Reply:
https://github.com/redpanda-data/redpanda/issues/8184